src/modifierGolemancer.ts

```diff
@@ -0,0 +1,124 @@
+import { registerEvents, registerModifiers } from "./cards";
+import { getOrInitModifier, Modifier } from "./cards/util";
+import * as Unit from './entity/Unit';
+import Underworld from './Underworld';
+import * as Upgrade from './Upgrade';
+import { golem_unit_id } from "./entity/units/golem";
+import seedrandom from "seedrandom";
+import { Vec2 } from "./jmath/Vec";
+import { prng, randFloat } from "./jmath/rand";
+import { COLLISION_MESH_RADIUS } from "./config";
+import { allUnits } from "./entity/units";
+import { UnitType } from "./types/commonTypes";
+import { makeRisingParticles } from "./graphics/ParticleCollection";
+import { BLOOD_GOLEM_ID } from "./entity/units/bloodGolem";
+
+// Summon up to [quantity] golems each turn
+export const golemancerId = 'Golemancer';
+export default function registerGolemancer() {
+  registerModifiers(golemancerId, {
+    description: 'rune_golemancer',
+    costPerUpgrade: 200,
+    add: (unit: Unit.IUnit, underworld: Underworld, prediction: boolean, quantity: number = 1) => {
+      getOrInitModifier(unit, golemancerId, { isCurse: false, quantity, keepOnDeath: true }, () => {
+        Unit.addEvent(unit, golemancerId);
+        const player = underworld.players.find(p => p.unit == unit);
+        if (player) {
+          // Give summon golem
+          const newCardId = Unit.unitSourceIdToName(golem_unit_id, false);
+          const upgrade = Upgrade.upgradeCardsSource.find(u => u.title == newCardId)
+          if (upgrade) {
+            underworld.forceUpgrade(player, upgrade, true);
+          } else {
+            console.error('Could not find summon golem upgrade for Golemancer rune');
+          }
+        }
+      });
+    },
+  });
+  registerEvents(golemancerId, {
+    onTurnStart: async (unit: Unit.IUnit, underworld: Underworld, prediction: boolean) => {
+      const modifier = unit.modifiers[golemancerId];
+      if (modifier && unit.alive) {
+        spawnGolems(unit, modifier.quantity, underworld, prediction);
+      }
+    },
+    onSpawn: (unit: Unit.IUnit, underworld: Underworld, prediction: boolean) => {
+      const modifier = unit.modifiers[golemancerId];
+      if (modifier && unit.alive) {
+        spawnGolems(unit, modifier.quantity, underworld, prediction);
+      }
+    }
+  });
+}
+
+function spawnGolems(unit: Unit.IUnit, quantity: number, underworld: Underworld, prediction: boolean) {
+  let allyGolems = getLivingAllyGolems(unit, underworld, prediction);
```

Comment:
I think the description is confusing me, so it only spawns up to quantity of golems but won't go beyond?

Reply:
Yes. If you have golemancer 5, and start your turn with 2 golems, it'll summon 3 more. If you start your turn with 5+, it wont summon any